SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialActorChannel.cpp

```diff
@@ -1172,11 +1163,33 @@ void USpatialActorChannel::UpdateSpatialPosition()
 			return;
 		}
 	}
+	
+	// Check that the Actor has satisfied both minimum thresholds OR either of the maximum thresholds
+	bool bSatisifiedPositionUpdateRequirements = false;
+
+	const float TimeSinceLastPositionUpdate = NetDriver->GetElapsedTime() - TimeWhenPositionLastUpdated;
 
-	// Check that the Actor has moved sufficiently far to be updated
-	const float SpatialPositionThresholdSquared = FMath::Square(GetDefault<USpatialGDKSettings>()->PositionDistanceThreshold);
 	FVector ActorSpatialPosition = SpatialGDK::GetActorSpatialPosition(Actor);
-	if (FVector::DistSquared(ActorSpatialPosition, LastPositionSinceUpdate) < SpatialPositionThresholdSquared)
+	const float DistanceTravelledSinceLastUpdateSquared = FVector::DistSquared(ActorSpatialPosition, LastPositionSinceUpdate);
```

Comment:
If always calculating this (which might be the right thing to do) then it's probably worth earlying out if the distance travelled is 0. As there is no case where you still want to update then and it's likely to be a common case.

Reply:
Yeah, I thought that even if we have travelled 0, we might still want to update after more than MaxTime elapsed, but I think what I thought had no sense, will change this, thank you for pointing it out!